homeassistant/components/media_player/epson.py

```diff
@@ -0,0 +1,316 @@
+ATTR_CMODE = 'cmode'
+SUPPORT_CMODE = 33001
+ACCEPT_ENCODING = "gzip, deflate"
+ACCEPT_HEADER = "application/json, text/javascript"
+EPSON_ERROR_CODE = 'ERR'
+
+SUPPORT_EPSON = SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_SELECT_SOURCE |\
+            SUPPORT_CMODE | SUPPORT_VOLUME_MUTE | SUPPORT_VOLUME_STEP | \
+            SUPPORT_NEXT_TRACK | SUPPORT_PREVIOUS_TRACK
+
+EPSON_SCHEMA = MEDIA_PLAYER_SCHEMA.extend({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Optional(ATTR_CMODE): cv.string
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the Epson media player platform."""
+    if DATA_EPSON not in hass.data:
+        hass.data[DATA_EPSON] = []
+
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    epson = EpsonProjector(
+        hass, name, host,
+        config.get(CONF_PORT), config.get(CONF_SSL), KEY_COMMANDS)
+
+    epson.update()
+    hass.data[DATA_EPSON].append(epson)
+    async_add_devices([epson], update_before_add=True)
+
+    @asyncio.coroutine
+    def async_service_handler(service):
+        """Handle for services."""
+        entity_ids = service.data.get('entity_id')
+        if entity_ids:
+            devices = [device for device in hass.data[DATA_EPSON]
+                       if device.entity_id in entity_ids]
+        else:
+            devices = hass.data[DATA_EPSON]
+        for device in devices:
+            if service.service == ATTR_CMODE:
+                cmode = service.data.get(ATTR_CMODE).lower()
+                if cmode in CMODE_LIST_SET:
+                    yield from device.select_cmode(cmode)
+            yield from device.update()
+    hass.services.async_register(
+        DOMAIN, ATTR_CMODE, async_service_handler,
+        schema=EPSON_SCHEMA)
+    return True
+
+
+class EpsonProjector(MediaPlayerDevice):
+    """Representation of Epson Projector Device."""
+
+    def __init__(self, hass, name, host, port, encryption, key_commands):
+        """Initialize entity to control Epson projector."""
+        self._hass = hass
+        self._name = name
+        self._host = host
+        self._port = port
+        self._cmode = None
+        self._source_list = list(DEFAULT_SOURCES.values())
+        self.key_commands = key_commands
+        self._encryption = encryption
+        self._state = STATE_UNKNOWN
+        http_protocol = 'https' if self._encryption else 'http'
+        self._http_url = '{http_protocol}://{host}:{port}/cgi-bin/'.format(
+            http_protocol=http_protocol,
+            host=self._host,
+            port=self._port)
+        referer = "{http_protocol}://{host}:{port}/cgi-bin/webconf".format(
+            http_protocol=http_protocol,
+            host=self._host,
+            port=self._port)
+        self._headers = {
+            "Accept-Encoding": ACCEPT_ENCODING,
+            "Accept": ACCEPT_HEADER,
+            "Referer": referer
```

Comment:
Please use the constants from `from aiohttp import hdrs` or `const.py` if possible.